runtime/vm/resolvefield.cpp

```diff
@@ -635,13 +635,16 @@ UDATA
 findIndexInFlattenedClassCache(J9FlattenedClassCache *flattenedClassCache, J9ROMNameAndSignature *nameAndSignature)
```

Comment:
`findJ9ClassInFlattenedClassCache` can also be updated to use the `J9RomFieldShape` as the search criteria

Reply:
@tajila When searching for the index, can I directly compare two J9RomFieldShapes to determine if they are the same? Or still get the NaS from each J9RomFieldShape and compare the NaS?

Reply:
@tajila Also J9ROMFieldShape *field has not been not initialized yet, what target J9RomFieldShape should be searched for?

Reply:
It looks like fieldshape is not available when the first time `findJ9ClassInFlattenedClassCache` is called, so I created two functions, `findJ9ClassInFCCFromNameAndSignature` and `findJ9ClassInFCCFromFieldShape`.